### PR TITLE
FISH-6045 Port gRPC to Payara 6

### PR DESCRIPTION
**Pull request by luiseufrasio**
## Description
Payara6 version of gRPC

### Blockers
None

### New tests
None

### Testing Performed
Payara-Community-Documentation\docs\modules\ROOT\pages\documentation\extensions\grpc\README.adoc

### Testing Environment
JDK "11.0.10" 2021-01-19 LTS on Windows 10 with Maven 3.8.4

### CODE REVIEW

File: grpc-support/src/main/java/fish/payara/extension/grpc/servlet/ServletServerStream.java [hunk 1]
```diff
@@ -17,41 +17,33 @@
...
 import io.grpc.Status;
 import io.grpc.Status.Code;
-import io.grpc.internal.AbstractServerStream;
-import io.grpc.internal.GrpcUtil;
-import io.grpc.internal.SerializingExecutor;
-import io.grpc.internal.StatsTraceContext;
-import io.grpc.internal.TransportFrameUtil;
-import io.grpc.internal.TransportTracer;
-import io.grpc.internal.WritableBuffer;
+import io.grpc.internal.*;
```
Comment (breakponchito) [inline grpc-support/src/main/java/fish/payara/extension/grpc/servlet/ServletServerStream.java]: it is preferable to not use the .* please add all the imports

File: grpc-support/src/main/java/fish/payara/extension/grpc/servlet/ServletServerStream.java [hunk 2]
```diff
@@ -17,41 +17,33 @@
...
-import javax.servlet.AsyncContext;
-import javax.servlet.WriteListener;
-import javax.servlet.http.HttpServletResponse;
+
+import static io.grpc.internal.GrpcUtil.CONTENT_TYPE_GRPC;
+import static io.grpc.internal.GrpcUtil.CONTENT_TYPE_KEY;
+import static java.lang.Math.max;
+import static java.lang.Math.min;
+import static java.util.Arrays.copyOfRange;
+import static java.util.logging.Level.*;
```
Comment (breakponchito) [inline grpc-support/src/main/java/fish/payara/extension/grpc/servlet/ServletServerStream.java]: same here

**Review by breakponchito — APPROVED** [review]
LGTM

File: pom.xml [hunk 3]
```diff
@@ -53,13 +53,13 @@
...
-        <maven.compiler.target>8</maven.compiler.target>
+        <maven.compiler.source>11</maven.compiler.source>
+        <maven.compiler.target>11</maven.compiler.target>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <grpc.version>1.43.1</grpc.version>
+        <grpc.version>1.45.0</grpc.version>
         <protobuf.version>3.19.4</protobuf.version>
-        <payara.version>5.2022.1</payara.version>
-        <payara.internal.version>5.2020.5</payara.internal.version>
+        <payara.version>6.2022.1.Alpha3-SNAPSHOT</payara.version>
```
Comment (breakponchito) [inline pom.xml]: Should we need to consider to create a new release of the internal dependencies? I think for now you are using the SNAPSHOT but we need to upgrade in some moment with the final release

Comment (Pandrex247) [reply pom.xml]: Yes ideally we would be using a final version.
It only appears to be pulling in the bom and no other internal dependencies, so we can presumably just set this to 6.2022.1.Alpha2?

**Pull request merged**